rts/motoko-rts/src/types.rs

```diff
@@ -540,31 +528,35 @@ impl Concat {
     }
 }
 
-#[repr(packed)]
 pub struct Null {
     pub header: Obj,
 }
 
-#[repr(packed)]
 pub struct Bits64 {
     pub header: Obj,
-    pub bits: u64,
+    // We have two 32-bit fields instead of one 64-bit to avoid aligning the fields on 64-bit
+    // boundary.
+    bits_hi: u32,
+    bits_lo: u32,
```

Comment:
An alternative here might be to use `[u8; 8]`, and then do the load with something like `bits.as_ptr().cast::<u64>().read_unaligned()`; that should translate to a single wasm load.

Reply:
> I am a bit surprised to see a new getter but not a new setter/constructor that properly encodes data into these fields

These objects are allocated by the generated code and the field is [written using Wasm `i64.store`][1]. We don't need to allocate them in the RTS so we don't have setter or allocation function for it.

Wasm is little endian so we have low bits first, then high bits.

[1]: https://github.com/dfinity/motoko/blob/cd0c3f8b04cf10361e6df7aef476bdf00caaa139/src/codegen/compile.ml#L1484